segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/AsyncStorageWrapper.java

```diff
@@ -138,6 +139,11 @@ public boolean supportsTruncation() {
         return supplyAsync(() -> this.syncStorage.exists(streamSegmentName), streamSegmentName);
     }
 
+    @Override
+    public CompletableFuture<Iterator<SegmentProperties>> listSegments() {
```

Comment:
So ... why is this returning a `CompletableFuture` if you are always wrapping a sync call?

Just return `Iterator<SegmentProperties` from the method.